### ISSUE TITLE
swapchain recreation error

### ISSUE THREAD

**Issue by AkumaRRR** [KhronosGroup/Vulkan-Samples#725]
```
Program: ...ild\windows\app\bin\debug\AMD64\vulkan_samples.exe
File: C:\Users\AkumaRRR\Desktop\Vulkan-Sample\framework\vulkan_sample.cpp
Line: 568

Expression: render_context && "Render context is not valid"
```

**Comment by SaschaWillems** [comment]
Can you elaborate? Hard to help with so little information.

- What did you do to trigger the error?
- What platform?
- What implementation (GPU, Vendor, driver)?

**Comment by AkumaRRR** [comment]
I simply run the sample in debug mode and this is the result:

![My Remote Image](https://i.ibb.co/nDsp1Nq/1.png)

Win10 22H2
RTX 3060 Ti (driver 531.14)

```
ucrtbased.dll!common_assert_to_message_box<wchar_t>(const wchar_t * const expression, const wchar_t * const file_name, const unsigned int line_number, void * const return_address) Line 388	C++
ucrtbased.dll!common_assert<wchar_t>(const wchar_t * const expression, const wchar_t * const file_name, const unsigned int line_number, void * const return_address) Line 424	C++
ucrtbased.dll!_wassert(const wchar_t * expression, const wchar_t * file_name, unsigned int line_number) Line 444	C++
vulkan_samples.exe!vkb::VulkanSample::get_render_context() Line 568	C++
vulkan_samples.exe!vkb::Platform::update() Line 199	C++
vulkan_samples.exe!vkb::Platform::main_loop() Line 152	C++
vulkan_samples.exe!WinMain(HINSTANCE__ * hInstance, HINSTANCE__ * hPrevInstance, char * lpCmdLine, int nCmdShow) Line 54	C++


```

**Comment by AkumaRRR** [comment]
All other samples work fine.

**Comment by tomadamatkinson** [comment]
@SaschaWillems i likely know the offending line will test and submit a patch for this

This sample may not create a render context in the same way as other samples

But the offender is the on post draw hook in platform

**Comment by tomadamatkinson** [comment]
I reproduced this bug and fixed it with #726

**Comment by AkumaRRR** [comment]
Thx!
